packages/uniforms/__suites__/ListField.tsx

```diff
@@ -0,0 +1,106 @@
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import createContext, { render } from './renderWithContext';
+
+function getDefaultContext() {
+  return createContext({ x: { type: Array }, 'x.$': { type: String } });
+}
+
+export function ListFieldTests(ListField: React.ComponentType<any>) {
+  test('<ListField> - renders ListAddField', () => {
+    render(<ListField name="x" label="ListFieldLabel" />, getDefaultContext());
+
+    expect(screen.getByRole('button')).toBeInTheDocument();
+  });
+
+  test('<ListField> - renders correct label (specified)', () => {
+    render(<ListField name="x" label="ListFieldLabel" />, getDefaultContext());
+
+    // Throws error with no form control found when getting by label
```

Comment:
Error is caused due to bad accessibility in this component so it would have to be fixed before we could change the component.